easybuild/easyconfigs/o/OTF2/OTF2-2.1.1-GCCcore-6.4.0.eb

```diff
@@ -0,0 +1,58 @@
+##
+# This is an easyconfig file for EasyBuild, see https://github.com/easybuilders/easybuild
+# Copyright:: Copyright 2013-2018 Juelich Supercomputing Centre, Germany
+# Authors::   Bernd Mohr <b.mohr@fz-juelich.de>
+#             Markus Geimer <m.geimer@fz-juelich.de>
+# License::   3-clause BSD
+#
+# This work is based on experiences from the UNITE project
+# http://apps.fz-juelich.de/unite/
+##
+
+easyblock = 'EB_Score_minus_P'
+
+name = 'OTF2'
+version = '2.1.1'
+
+homepage = 'http://www.score-p.org'
+description = """
+ The Open Trace Format 2 is a highly scalable, memory efficient event trace
+ data format plus support library. It is the new standard trace format for
+ Scalasca, Vampir, and TAU and is open for other tools.
+
+"""
+
+toolchain = {'name': 'GCCcore', 'version': '6.4.0'}
+
+source_urls = ['http://www.vi-hps.org/upload/packages/otf2/']
+sources = [SOURCELOWER_TAR_GZ]
+checksums = [
+    '01591b42e76f396869ffc84672f4eaa90ee8ec2a8939755d9c0b5b8ecdcf47d3',  # otf2-2.1.1.tar.gz
+]
+
+builddependencies = [
+    # use same binutils version that was used when building GCCcore
+    ('binutils', '2.28'),
+    # SIONlib container support (optional):
+    ('SIONlib', '1.7.2', '-tools'),
+]
+
+dependencies = [
+    # Python bindings (optional)
+    ('Python', '2.7.14', '-bare'),
+    ('future', '0.16.0', '-Python-2.7.14-bare'),
+    ('Six', '1.11.0', '-Python-2.7.14-bare'),
```

Comment:
@geimer By using a `-bare` Python here, you're effectively making this incompatible with anything that depends on a full `Python`, which is quite common... Isn't that a concern?

Maybe it makes sense to make the `Python` that gets built with `GCCcore` a trimmed down version of the one we have with a full toolchain, i.e. still include common dependencies like `six`, `future`, `setuptools`, `pip`, etc into it?

Reply:
@boegel There doesn't seem to be any consensus how to deal with `Python` on the `GCCcore` level, cfr. the discussion I raised two or three weeks ago on the mailing list. I'm doing *something* here to make progress, but the `Python` stuff needs to be fixed first, and then I can adjust this PR accordingly. But I'm **not** the one who is going to take a lead on this, as there are too many dark corners I don't understand...

Reply:
Good point... Maybe you can consider building the Python bindings in a separate module, which can be loaded as needed?

Reply:
Not sure our build system supports this right now.  Is there any support in EB for building things twice into the same prefix but with different build dependencies (like with a list of `configopts`)?  This may work too, but I would immediately admit that this is a rather strange feature request ;-)

Reply:
Talking to the OTF2 core devs, this currently seems to be the only option: building OTF2 twice into the same prefix with the different Python's as build dependencies, and then providing Python binding modules that simply depend on the common OTF2 installation and only set `PYTHONPATH` accordingly. But I don't think this is currently possible with EB...

Reply:
Not in a clean way, no, but it relates closely to the way in which the ComputeCanada folks install Python packages, so we should come up with a clean way of doing that anyway...

@bartoldeman @mboisson Do you have your approach clearly documented somewhere? Maybe it's time to start implementing this...

Reply:
What our approach boils down to:
1. Let the Python module set PYTHONPATH to a location with a file `sitecustomize.py`
2. this file has the following contents:
```
import sys
import os
import site

# use prefixes from EBPYTHONPREFIXES, so they have lower priority than
# virtualenv-installed packages

if "EBPYTHONPREFIXES" in os.environ:
    postfix = os.path.join('lib', 'python'+sys.version[:3], 'site-packages')
    for prefix in os.environ["EBPYTHONPREFIXES"].split(os.pathsep):
        sitedir = os.path.join(prefix, postfix)
        if os.path.isdir(sitedir):
            site.addsitedir(sitedir)
```
3. modules that use Python set EBPYTHONPREFIXES. We now do it as follows:
```
modextrapaths = {
    'EBPYTHONPREFIXES': [''],
}
```
4. module have to load different python modules for builds. We now do this as follows:
```
prebuildopts = ['module load python/2.7 && ',
                'module load python/3.5 && ',
                'module load python/3.6 && ']
```
but that's a hack. It would be better to allow builddependencies to be a list of lists.
5. we have this for some modules that absolutely need python (for some it is optional):
modluafooter = """
depends_on("python")
"""
this basically will load the default python if none is loaded and otherwise keep whatever python module is already loaded.

Hope that helps.